### PR TITLE
Uses websocket to get sensor device class units

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Fetch sensor device class units from core instead of hardcoded ones.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion: #https://github.com/home-assistant/core/pull/85213
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/entities/entity-registry-settings.ts [hunk 1]
```diff
@@ -326,6 +277,20 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
...
+  protected async updated(changedProps: PropertyValues): Promise<void> {
+    if (changedProps.has("_deviceClass")) {
+      const domain = computeDomain(this.entry.entity_id);
+
+      if (domain === "sensor" && this._deviceClass) {
+        const { units } = await getSensorDeviceClassUnits(
+          this.hass,
+          this._deviceClass
+        );
+        this._sensorDeviceClassUnits = units;
```
[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/config/entities/entity-registry-settings.ts:289]: So we need to hardcode the convertible unit list.
For example, `frequency` device_class has multiple units available but isn't not convertible (for now). So I shouldn't be changed from the UI.

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/entities/entity-registry-settings.ts [hunk 2]
```diff
@@ -177,6 +126,24 @@ const OVERRIDE_WEATHER_UNITS = {
   wind_speed: ["ft/s", "km/h", "kn", "m/s", "mph"],
 };
 
+const CONVERTIBLE_UNITS_SENSOR_DEVICE_CLASS = new Set([
```
Comment (bramkragten) [inline src/panels/config/entities/entity-registry-settings.ts:129]: Can we include this in the ws command from core?

File: src/data/sensor.ts [hunk 3]
```diff
@@ -1,2 +1,15 @@
+import { HomeAssistant } from "../types";
+
 export const SENSOR_DEVICE_CLASS_BATTERY = "battery";
 export const SENSOR_DEVICE_CLASS_TIMESTAMP = "timestamp";
+
+export type SensorDeviceClassUnits = { units: string[] };
```
Comment (bramkragten) [inline src/data/sensor.ts:6]: Something like
```suggestion
export type SensorDeviceClassUnits = { units: string[]; convertible: boolean; };
```

Comment (piitaya) [reply src/data/sensor.ts:6]: That's what I suggested in the core PR

Comment (epenet) [reply src/data/sensor.ts:6]: Is there a plan to make use of this list for things other than unit conversion?
If it's only for unit conversion, then I think it's better to simply return an empty list when it's not convertible, as adjusted in https://github.com/home-assistant/core/pull/85213/commits/96321584a208b6d79d638eb01565001d09043662.

Comment (piitaya) [reply src/data/sensor.ts:6]: I updated the endpoint et removed the hardcoded list in frontend

**Review by balloob — APPROVED** [review]

**Pull request merged**